src/Nethermind/Nethermind.Consensus/Scheduler/BackgroundTaskScheduler.cs

```diff
@@ -126,6 +126,11 @@ public void ScheduleTask<TReq>(TReq request, Func<TReq, CancellationToken, Task>
 
         if (!_taskQueue.Writer.TryWrite(activity))
         {
+            if (request is IDisposable disposable)
```

Comment:
This risk double dispose. Don't think this is a good idea. The FullfillFunc should own the request now and dispose it.

Reply:
1. This is on fail write, so FullfillFunc will not be called
2. Dispose should be able to be called multiple times - this is in .net spec: https://stackoverflow.com/a/5306896/1187056